typescript_templates/algod_config.properties

```diff
@@ -7,3 +7,54 @@
 #     changes.
 use_object_params_if_greater_than=0
 indexer=false
+# Type override conventions:
+# * All Algo and asset amounts should be bigint (including fees)
+# * All round numbers should be bigint, including any quantities you would add
+#   to a round number (e.g. offset or duration)
+# * All application and asset IDs should be bigint
+# * All timestamps in nanoseconds should be bigint
+# * If there is a quantity whose purpose is to be added to, subtracted from, or
+#   otherwise operated on with a bigint, it should be a bigint for ease of use
+# * Other quantities outside of the above and which will never exceed the maximum
+#   safe javascript integer should be number
+type_override_Account_amount=bigint
```

Comment:
Dumb question, why should we leave this as configurable at all? While we are putting sane defaults here, what are the cases where we should support the client overriding?

Reply:
First I should explain what's happening in this file. It's really just a list of key-value definitions.

In the template code, specifically here: https://github.com/algorand/generator/blob/4b0120a3850e3fd58c0412e767e4f59f684c27ef/typescript_templates/model.vm#L7-L10

we look at `propFile.type_override_X_Y`, where `X` is the generated class name and `Y` is the field in that class. If there's a value associated with that key from the prop file, we will use the type override that it declares.

So it's less about making each field configurable, and more about configuring the list of fields and their overrides.

I chose to put the overrides in this config file for a few reasons:
1. Different fields needed to be overriden for algod and indexer
2. It's much easier to encode this information in a file like this instead of placing a massive if/elseif/.../else statement in the template code